ui/app/components/brave/brave-header/brave-header.component.js

```diff
@@ -0,0 +1,75 @@
+import React, { PureComponent } from 'react'
+
+export default class BraveHeader extends PureComponent {
```

Comment:
is this header re-used in other Brave pages? maybe it can be split into an npm module and imported here if so?

Reply:
@diracdeltas it's implemented in core as a Polymer template so I had to make some modifications:

https://github.com/brave/brave-core/blob/master/ui/webui/resources/br_elements/br_toolbar/br_toolbar.html

Also I need to add some JS to this to handle the chrome links as those will not work as they are inside an extension window

I think it could be worth standardizing eventually but it may be too much of an undertaking in the context of this PR

Reply:
> Also I need to add some JS to this to handle the chrome links as those will not work as they are inside an extension window

got it. do you have a plan for this? for security reasons i think non-chrome:// protocol pages cannot open chrome:// pages

Reply:
@diracdeltas using the tabs api worked out well :) - https://github.com/brave/ethereum-remote-client/pull/2/commits/c7a441481ae3427abbe29e77f999891c9fe63769

Curiously something like `window.location.href = 'chrome://page'` throws an error, so I guess the tabs API makes this acceptable? 🤷‍♂️

Reply:
the `window.location` update will fail in webpages as well so appears to be a special privilege for chrome extensions

Reply:
makes sense since extensions have to opt into the `tabs` permission, IIRC